src/pytom_tm/parallel.py

```diff
@@ -13,7 +14,13 @@
     pass
 
 
-def gpu_runner(gpu_id: int, task_queue: BaseProxy, result_queue: BaseProxy, log_level: int) -> None:
+def gpu_runner(
+        gpu_id: int, task_queue: BaseProxy, result_queue: BaseProxy, log_level: int, mute: bool
+) -> None:
+    if mute:
+        devnull = open(os.devnull, 'w')
+        sys.stdout = devnull
+        sys.stderr = devnull
```

Comment:
this breaks any further writes to `sys.stdout` and `sys.stderr` by any further code.
it is probably better to make a mute `contextmanager` (following [this so answer](https://stackoverflow.com/a/52442331/21764481)) :
```python
from contextlib import contextmanager,redirect_stderr,redirect_stdout
from os import devnull

@contextmanager
def mute_stdout_stderr():
    """A context manager that redirects stdout and stderr to devnull"""
    with open(devnull, 'w') as fnull:
        with redirect_stderr(fnull) as err, redirect_stdout(fnull) as out:
            yield (err, out)
```
And then in this code:
```python
if mute:
    mute_context = mute_stdout_stderr() #Make sure it is imported at this time
else:
    mute_context =  contextlib.nullcontext()
with mute_context:
    logging.basicConfig(level=log_level)
    while True:
        ...
```

Reply:
that is much nicer, but I am still getting this big error because of the invalid device I am passing

```
Process Process-3:
Traceback (most recent call last):
  File "/home/marten/miniconda3/envs/pytom_tm/lib/python3.11/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/home/marten/miniconda3/envs/pytom_tm/lib/python3.11/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/marten/coding/pytom-template-matching-gpu/src/pytom_tm/parallel.py", line 30, in gpu_runner
    result_queue.put_nowait(job.start_job(gpu_id, return_volumes=False))
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marten/coding/pytom-template-matching-gpu/src/pytom_tm/tmjob.py", line 467, in start_job
    tm = TemplateMatchingGPU(
         ^^^^^^^^^^^^^^^^^^^^
  File "/home/marten/coding/pytom-template-matching-gpu/src/pytom_tm/matching.py", line 74, in __init__
    cp.cuda.Device(device_id).use()
  File "cupy/cuda/device.pyx", line 192, in cupy.cuda.device.Device.use
  File "cupy/cuda/device.pyx", line 198, in cupy.cuda.device.Device.use
  File "cupy_backends/cuda/api/runtime.pyx", line 375, in cupy_backends.cuda.api.runtime.setDevice
  File "cupy_backends/cuda/api/runtime.pyx", line 144, in cupy_backends.cuda.api.runtime.check_status
cupy_backends.cuda.api.runtime.CUDARuntimeError: cudaErrorInvalidDevice: invalid device ordinal
```

Reply:
Yeah, reading up on it it seems like that function does explicitly not mute raised errors. 
You can write your own context as a class with `__enter__` and `__exit__` functions and use that. Currently on mobile, but let me know if you want me to write a complete example class (will probably then do it tomorrow)

Reply:
I pushed a commit with a context manager like this (took some inspiration from an answer in the SO thread). Seems to work now. I left out all os.dup2 that this answer did (https://stackoverflow.com/a/50438156), any idea what that does?

Reply:
I think as I have it now it should be fine for the tests